classes/migration/upgrade/v3_4_0/I8060_UpdateUserLocalesDefaultToEmptyArrayFromNull.inc.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I8060_UpdateUserLocalesDefaultToEmptyArrayFromNull.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I8060_UpdateUserLocalesDefaultToEmptyArray
+ * @brief Update the users table locales column default to empty array from NULL and update existing NULL ones to []
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+class I8060_UpdateUserLocalesDefaultToEmptyArrayFromNull extends Migration
+{
+    public function up(): void
+    {
+        DB::table('users')->whereNull('locales')->update(['locales' => '[]']);
+
+        Schema::table('users', function ($table) {
+            $table->string('locales', 255)->nullable(false)->default('[]')->change();
+        });
+    }
+
+    public function down(): void
+    {
+        throw new DowngradeNotSupportedException('Downgrade unsupported due to updated data');
```

Comment:
Basically, we can support downgrades in these cases by saving data as a property of the current migration object before we modify it.

Reply:
In this case, I would recommend restoring the `locales` column to `nullable` but leaving the data as it was after the `up` function runs. The difference between `null` and `'[]'` is negligible, so not worth the pain of implementing a 100% restore to the previous state.

But yes, @Vitaliy-1, in theory that's possible -- I hadn't considered it and will probably use it in other migrations!

Reply:
@asmecher I have update the migration `down` so that it can be reverted back to `nullable` but I think it' best to keep the `default` to `[]` .